### ISSUE TITLE
Cannot connect to database server

### ISSUE THREAD

**Issue by hainguyen291** [nordic-institute/X-Road#21]
Hi everyone,

I am working on `X-Road: External Load Balancer Installation Guide` [X-Road: External Load Balancer Installation Guide](https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md#x-road-external-load-balancer-installation-guide) on `master` server. I restarted all xroad services again. Although I still can login to SS via port `4000`, I cannot connect to database server anymore. 

`gov@ubuntu:~$ sudo service postgresql status`
`9.3/main (port 5432): online`
`9.3/serverconf (port 5433): online`
`gov@ubuntu:~$`

![image](https://user-images.githubusercontent.com/12641471/45406821-d8e1f400-b691-11e8-84ca-13b5b94952a6.png)

I read `External Load Balancer` again to make sure didn't do anything wrong. But I couldn't find any missing steps. Please let me know what I should do now.

Best regards,
Hai

**Comment by petkivim** [comment]
Hi @hainguyen291 

Please verify that you have completed [chapter 3.2 step 4](https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md#32-master-installation) and restarted services after that.

> Change `/etc/xroad/db.properties` to point to the separate database instance:
>
>    `serverconf.hibernate.connection.url` : Change the url port number from `5432` to `5433` (or the port you specified)

Regards,
Petteri

**Comment by hainguyen291** [comment]
I am sorry for late communication, I were quite busy last week. I will check it out every step again. BTW, are there any MTU limitation of Security Server? I could send a SOAP request with 30MB attachment successfully,  however it didn't work with a file more than 50Mb. I always get a response such as: connection refused from SS.

**Comment by petkivim** [comment]
Hi @hainguyen291 

There are no hard limits regarding the size of messages and/or attachments that the Security Server can process. The maximum size depends on the memory allocated for `xroad-proxy` process and available disk space. SOAP attachments are written temporarily on disk so there must be enough free disk space when sending big attachments. In your case the problem is probably insufficient Java heap size for `xroad-proxy` process. You can find instructions for increasing the heap size here:

https://confluence.niis.org/display/XRDKB/Recommended+Memory+Allocation+for+Proxy

Regards,
Petteri

**Comment by hainguyen291** [comment]
Hi @petkivim 

Thanks for your information.

Regards,
Hai.

**Comment by hainguyen291** [comment]
I am going to configure slave instance for replication. However, I always get the error when tried to run the following command:

>sudo -u postgres PGSSLMODE=verify-ca PGSSLROOTCERT=/etc/xroad/postgresql/ca.crt PGSSLCERT=/etc/xroad/postgresql/server.crt PGSSLKEY=/etc/xroad/postgresql/server.key pg_basebackup -h <master> -p 5433 -U <nodename> -D .

![image](https://user-images.githubusercontent.com/12641471/45805552-2f31e100-bce9-11e8-8213-63c6f6b1f1a7.png)

My master server is located on 192.168.132.132 and I tried to add slave01 user to master db. It seems that there are something wrong with my CA signature on step [4.1 Setting up TLS certificates for database authentication](https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md#41-setting-up-tls-certificates-for-database-authentication)
I should use the same ca.key and ca.crt generated on master server to generate server.key and server.crt for slave01 server, shouldn't I?

Best regards,
Hai

**Comment by qrlt** [comment]
Yes, You should use the same CA key and cert in master instance to generate key and csr for the slave instance.

The connection problem, please make sure that the slave nodename is the same in all these commands(for ex. I used a nodename "slave1"):

Generating key for slave(in master machine):
`$ openssl req -new -nodes -days 7300 -keyout slave1.key -out slave1.csr -subj "/O=cluster/CN=slave1"`

Creating the replication user(in master machine):
`$ sudo -u postgres psql -p 5433 -c "CREATE USER slave1 REPLICATION PASSWORD NULL IN ROLE slavenode";`

Base backup(in slave machine):
`$ sudo -u postgres PGSSLMODE=verify-ca PGSSLROOTCERT=/etc/xroad/postgresql/ca.crt PGSSLCERT=/etc/xroad/postgresql/server.crt PGSSLKEY=/etc/xroad/postgresql/server.key pg_basebackup -h karlt-master.ci.kit -p 5433 -U slave1 -D .`


Also, make sure Your slave machine cert and key filenames match in the base backup command and in /etc/xroad/postgresql/.

**Comment by hainguyen291** [comment]
Hi @qrlt 

I did generate slave01.key and slave01.crt in master machine. Then I copy there files including ca.crt (root certificate in master machine) onto slave machine on `/etc/xroad/postgresql/` directory. I change `r` permission to this folder and made postgre user own them. I also verified these slave01.crt and slave01.key

![image](https://user-images.githubusercontent.com/12641471/45833803-cde13080-bd2f-11e8-9979-a2ec935aa735.png)

![image](https://user-images.githubusercontent.com/12641471/45833867-fcf7a200-bd2f-11e8-9300-eea40898c5e0.png)

I added slave01 onto master database
![image](https://user-images.githubusercontent.com/12641471/45834057-73949f80-bd30-11e8-9817-275dd58ebebe.png)

However, I still get the above error
![image](https://user-images.githubusercontent.com/12641471/45833907-100a7200-bd30-11e8-8403-24a37d6b34c0.png)

Best regards,
Hai

I don't know why
